transport/EndpointFactoryImpl.cpp

```diff
@@ -0,0 +1,26 @@
+#include "transport/EndpointFactoryImpl.h"
+#include <memory>
+
+namespace transport
+{
+IUdpEndpoint* EndpointFactoryImpl::createUdpEndpointStatic(jobmanager::JobManager& jobManager,
```

Comment:
Personally I would prefer factories to return unique_ptr instead of raw pointers. Even if you need to store a shared_ptr later, you can move ownership of unique_ptr into a shared_ptr

Reply:
The real reason behind returning raw pointer is that we're using custom deleter in the TransportFactory. I found no way how to "append" a custom deleter to existing shared pointer.  And extracting raw from shared only to put it back to another shared, but this time with a custom deleter looks wrong.